File: WebClientIS/WebClientIS/Models/Relatorio.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace APIIS.Models
{
    public class Relatorio
    {
        public DateTime DataRelatorio { get; set; }

        public int IdP { get; set; }

        public int IdA { get; set; }

        public byte[] Ficheiro { get; set; }


        public Relatorio()
        {
            DataRelatorio = new DateTime();
            IdP = 0;
            IdA = 0;
            Ficheiro = null;
        }

        

        public List<Relatorio> GetRelatorios(int id,string what, object value)
        {
            List<Relatorio> relatorios = new List<Relatorio>();

            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Relatorio/" + id.ToString() + "/"+what+"/"+value.ToString());

            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

            HttpResponseMessage response = client.GetAsync("").Result;

            if (response.IsSuccessStatusCode)
            {
                var products = response.Content.ReadAsStringAsync().Result;

                var array = JArray.Parse(products);

                foreach (var obj in array)
                {
                    try
                    {
                        relatorios.Add(obj.ToObject<Relatorio>());
                    }
                    catch (Exception ex)
                    {
                        return new List<Relatorio>();
                    }
                }

            }

            return relatorios;
        }

        public bool CreateRelatorio(Relatorio rel)
        {
            HttpClient client = new HttpClient();

            var values = new Dictionary<string, string>
            {
                { "DataRelatorio", rel.DataRelatorio.ToString() },
                {"IdP",rel.IdP.ToString() },
                {"IdA",rel.IdA.ToString() },
                {"Ficheiro",rel.IdA.ToString() }


            };

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Relatorio");

            var jsonString = JsonConvert.SerializeObject(values);
            var content = new StringContent(jsonString, Encoding.UTF8, "application/json");

            var resposta = client.PostAsync(client.BaseAddress.ToString(), content).Result;

            if (resposta.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: MobileClient/sampleproject/lib/Models/Relatorio.dart
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert' show json;
import 'package:scoped_model/scoped_model.dart';


class Relatorio extends Model{

  DateTime datarelatorio;
  int idp;
  int ida;
  Uint8List ficheiro;


  Relatorio({DateTime dr, int idpp, int idaa, Uint8List file})
  {
    this.datarelatorio = dr;
    this.idp = idpp;
    this.ida = idaa;
    this.ficheiro = file;
  }

  factory Relatorio.fromJson(Map<String,dynamic> json){

     String bystring = json['Ficheiro'] as String;

     List<int> list = bystring.codeUnits;
     Uint8List bytes = Uint8List.fromList(list);

    return Relatorio(
      dr: DateTime.parse(json['DataRelatorio']),
      idpp: json['idP'] as int,
      idaa: json['idA'] as int,
      file: bytes
      
    );

  }

  Future<List<Relatorio>> getrelatorios() async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/Relatorio"),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<Relatorio> listau = new List<Relatorio>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(Relatorio.fromJson(lista[i]));
  

  return listau;
}

Future<List<Relatorio>> getrelatorio(int id,String what, String value) async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/Relatorio/"+id.toString()+"/"+what+"/"+value),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<Relatorio> listau = new List<Relatorio>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(Relatorio.fromJson(lista[i]));
  

  return listau;
}

Future<int> createRelatorio(Relatorio relatorio) async {

  var list  = new List.from(relatorio.ficheiro);

  var url ='http://6f85a52b4f05.ngrok.io/api/Relatorio';
  var body = json.encode(<String, dynamic>{
      'DataRelatorio': relatorio.datarelatorio.toString(),
      'IdP':relatorio.idp.toString(),
      'IdA':relatorio.ida.toString(),
      'Ficheiro':list
      
    });

  print(body);



   http.Response response = await http.post(url,
      headers: {"Content-Type": "application/json"},
      body: body
  );

  return response.statusCode;

}

}